controllers/common.go

```diff
@@ -57,6 +58,74 @@ func BuildCollector(params manifests.Params) ([]client.Object, error) {
 	return resources, nil
 }
 
+// @TODO: Later merge this in to reconcileDesiredObjects after implementing search functions for DCGM
+func reconcileDesiredObjectsWPrune(ctx context.Context, kubeClient client.Client, logger logr.Logger, owner v1alpha1.AmazonCloudWatchAgent, scheme *runtime.Scheme,
```

Comment:
Likewise, would it be plausible to include this function in a unit test to see if it appropriately deletes?

Reply:
There are no unit-test available for controllers, they are tested via integration tests